numpy/core/records.py

```diff
@@ -238,17 +237,14 @@ def __getattribute__(self, attr):
         res = fielddict.get(attr, None)
         if res:
             obj = self.getfield(*res[:2])
-            # if it has fields return a recarray,
-            # if it's a string ('SU') return a chararray
+            # if it has fields return a record,
             # otherwise return the object
             try:
                 dt = obj.dtype
             except AttributeError:
-                return obj
+                return obj #happens if field is Object type
```

Comment:
Please, no trailing comments except, perhaps, for C struct elements. Just put it on the previous line.